qutip/bloch.py

```diff
@@ -414,20 +418,21 @@ def run_from_ipython(self):
         except NameError:
             return False
 
-    def render(self, fig=None, axes=None):
+    def render(self):
         """
         Render the Bloch sphere and its data sets in on given figure and axes.
         """
-        if self._rendered:
-            self.axes.clear()
-
-        self._rendered = True
-
-        # Figure instance for Bloch sphere plot
-        if not fig:
+        if not self._ext_fig:
+            if self.fig is not None and not plt.fignum_exists(self.fig.number):
+                # a previous call to .rander() created the figure, but the
+                # figure not longer exists so we trigger recreating it:
+                self.fig = None
+                self.axes = None
+
+        if not self._ext_fig and self.fig is None:
             self.fig = plt.figure(figsize=self.figsize)
 
-        if not axes:
+        if not self._ext_axes and self.axes is None:
             self.axes = _axes3D(self.fig, azim=self.view[0],
                                 elev=self.view[1])
```

Comment:
What happen if `axes` are given but not `fig`? The `if not self._ext_fig:` would overwrite `self.axes` with `None` but they would not be recreated since `if not self._ext_axes and self.axes is None` will be `False`.

Reply:
Good point. Let me figure out what to do.